client/app/competitor/page.tsx

```diff
@@ -69,14 +77,14 @@ const EditorButtons = () => {
                     </Button>
                 </Tooltip>
                 <span className="ml-auto">
-                    <Select>
-                        <SelectTrigger className="w-56" defaultValue={currQuestion?.languages?.[0]}>
+                    <Select value={selectedLanguage} onValueChange={handleLanguageSelection}>
+                        <SelectTrigger className="w-56">
                             <SelectValue placeholder="Programming Language" />
                         </SelectTrigger>
                         <SelectContent>
                             {currQuestion?.languages?.map((l) => (
-                                <SelectItem key={l} value={l}>
-                                    {l}
+                                <SelectItem key={l.syntax} value={l.syntax}>
+                                    {l.syntax}
                                 </SelectItem>
```

Comment:
Making this all syntax is incorrect.  We have the distinction between a language name and its syntax.  Syntaxes aren't unique, but language names are.  Think of Python2 vs Python3.  The syntax for both is just "python", while the names differ.